src/pagination.rs

```diff
@@ -212,10 +256,80 @@ fn sort_posts(posts: &mut Vec<&liquid::value::Value>, config: &PaginationConfig)
     })
 }
 
+fn distribute_posts_by_tags<'a>(
+    all_posts: &[&'a liquid::value::Value],
+) -> HashMap<String, Vec<&'a liquid::value::Value>> {
+    let mut per_tags: HashMap<String, Vec<&'a liquid::value::Value>> = HashMap::new();
+    let add_to_tag =
+        |tag: &str,
+         post: &'a liquid::value::Value,
+         per_tags: &mut HashMap<String, Vec<&'a liquid::value::Value>>| {
+            if per_tags.contains_key(tag) {
+                let v = per_tags
+                    .get_mut(tag)
+                    .expect("Tag's presence has been tested, can't happen!");
+                v.push(post)
+            } else {
+                per_tags.insert(tag.to_string(), vec![post]);
```

Comment:
Can't you simplify this with the [`entry` API](https://doc.rust-lang.org/std/collections/struct.HashMap.html#method.entry)?